lms/djangoapps/certificates/generation_handler.py

```diff
@@ -210,35 +225,117 @@ def _can_generate_certificate_common(user, course_key):
         # The invalidation list prevents certificate generation
-        log.info(f'{user.id} : {course_key} is on the certificate invalidation list. Certificate cannot be generated.')
+        log.info(f'{user.id} : {course_key} is on the certificate invalidation list. Downloadable certificate cannot '
+                 f'be generated.')
         return False
 
     enrollment_mode, __ = CourseEnrollment.enrollment_mode_for_user(user, course_key)
     if enrollment_mode is None:
-        log.info(f'{user.id} : {course_key} does not have an enrollment. Certificate cannot be generated.')
+        log.info(f'{user.id} : {course_key} does not have an enrollment. Downloadable certificate cannot be generated.')
         return False
 
     if not modes_api.is_eligible_for_certificate(enrollment_mode):
         log.info(f'{user.id} : {course_key} has an enrollment mode of {enrollment_mode}, which is not eligible for a '
-                 f'certificate. Certificate cannot be generated.')
+                 f'certificate. Downloadable certificate cannot be generated.')
         return False
 
     if not IDVerificationService.user_is_verified(user):
-        log.info(f'{user.id} does not have a verified id. Certificate cannot be generated for {course_key}.')
+        log.info(f'{user.id} does not have a verified id. Downloadable certificate cannot be generated for '
+                 f'{course_key}.')
         return False
 
     if not _can_generate_certificate_for_status(user, course_key):
         return False
 
     course = _get_course(course_key)
     if not has_html_certificates_enabled(course):
-        log.info(f'{course_key} does not have HTML certificates enabled. Certificate cannot be generated for '
-                 f'{user.id}.')
+        log.info(f'{course_key} does not have HTML certificates enabled. Downloadable certificate cannot be generated '
+                 f'for {user.id}.')
         return False
 
     return True
 
 
+def _get_allowlist_cert_status(user, course_key):
+    """
+    Determine the allowlist certificate status for this user, in this course run.
+
+    This is used when a downloadable cert cannot be generated, but we want to provide more info about why it cannot
+    be generated.
+    """
+    if _is_cert_downloadable(user, course_key):
+        return None
+
+    if not is_using_certificate_allowlist(course_key):
+        return None
+
+    if not is_on_certificate_allowlist(user, course_key):
+        return None
+
+    course = _get_course(course_key)
+    return _get_cert_status_common(user, course_key, course)
+
+
+def _get_v2_cert_status(user, course_key):
+    """
+    Determine the V2 certificate status for this user, in this course run.
+
+    This is used when a downloadable cert cannot be generated, but we want to provide more info about why it cannot
+    be generated.
+    """
+    if _is_cert_downloadable(user, course_key):
```

Comment:
Can this check be added to the `_get_cert_status_common` function? It appears common to both `_get_allowlist_cert_status` and the `_get_v2_cert_status` functions.